dhcpv4/option_userclass.go

```diff
@@ -66,6 +66,17 @@ func ParseOptUserClass(data []byte) (*OptUserClass, error) {
 			totalLength, len(data))
 	}
 
+	// Check if option is Microsoft style instead of RFC compliant
+	var counting int
+	for counting < totalLength {
+		// UC_Len_i does not include itself so add 1
+		counting += int(data[counting]) + 1
+	}
+	if counting != totalLength {
+		opt.UserClasses = append(opt.UserClasses, data)
```

Comment:
I think this should be `opt.UserClasses = append(opt.UserClasses, data[:totalLength])`
Can a `TestParseOptUserClassMicrosoftLongerThanLength` be added?
I think it will fail if that is not changed.